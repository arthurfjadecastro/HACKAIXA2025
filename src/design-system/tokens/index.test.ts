import * as tokensIndex from './index';

describe('Design System Tokens Index', () => {
  it('should export all design tokens', () => {
    expect(tokensIndex).toBeDefined();
    expect(typeof tokensIndex).toBe('object');
  });

  it('should export colors', () => {
    expect(tokensIndex.colors).toBeDefined();
    expect(typeof tokensIndex.colors).toBe('object');
  });

  it('should export typography tokens', () => {
    expect(tokensIndex.typography).toBeDefined();
    expect(tokensIndex.fontWeights).toBeDefined();
    expect(tokensIndex.fontSizes).toBeDefined();
    expect(tokensIndex.lineHeights).toBeDefined();
  });

  it('should export spacing tokens', () => {
    expect(tokensIndex.spacing).toBeDefined();
    expect(tokensIndex.padding).toBeDefined();
    expect(tokensIndex.margin).toBeDefined();
    expect(tokensIndex.gap).toBeDefined();
  });

  it('should export radius tokens', () => {
    expect(tokensIndex.borderRadius).toBeDefined();
    expect(tokensIndex.radius).toBeDefined();
  });

  it('should export shadows and elevation', () => {
    expect(tokensIndex.shadows).toBeDefined();
    expect(tokensIndex.elevation).toBeDefined();
  });

  it('should export theme object', () => {
    expect(tokensIndex.theme).toBeDefined();
    expect(typeof tokensIndex.theme).toBe('object');
  });

  it('should match snapshot', () => {
    expect(Object.keys(tokensIndex).sort()).toMatchSnapshot();
  });
});
