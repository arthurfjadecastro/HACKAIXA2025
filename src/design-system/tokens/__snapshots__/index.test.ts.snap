// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Design System Tokens Index should match snapshot 1`] = `
[
  "borderRadius",
  "colors",
  "elevation",
  "fontSizes",
  "fontWeights",
  "gap",
  "lineHeights",
  "margin",
  "padding",
  "radius",
  "shadows",
  "spacing",
  "theme",
  "typography",
]
`;
